src/Microsoft.CodeQuality.Analyzers/UnitTests/ApiDesignGuidelines/AsyncMethodNamesShouldEndInAsyncTests.Fixer.cs

```diff
@@ -29,5 +30,391 @@ protected override CodeFixProvider GetCSharpCodeFixProvider()
         {
             return new CSharpAsyncMethodNamesShouldEndInAsyncFixer();
         }
+
+        private DiagnosticResult GetBasicResultAt(int line, int column, string methodName)
+        {
+            return GetBasicResultAt(line, column, AsyncMethodNamesShouldEndInAsyncAnalyzer.Rule, methodName);
+        }
+
+        private DiagnosticResult GetCSharpResultAt(int line, int column, string methodName)
+        {
+            return GetCSharpResultAt(line, column, AsyncMethodNamesShouldEndInAsyncAnalyzer.Rule, methodName);
+        }
+
+        private static string GetFullSource(string body)
+        {
+            return $@"
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ConsoleApplication
+{{
+    class Program
+    {{   {body}
+    }}
+}}
+";
+        }
+
+        // Async method does not includes "Async" - 1 diagnostic
+        [Fact]
+        public void AsyncMethodWithoutAsync()
+        {
+            var body = @"
+        private async Task<int> Math(int num1) 
+        { 
+            await Task.Delay(3000); 
+            return (num1); 
+        }";
+
+            var test = GetFullSource(body);
+
+            VerifyCSharp(test, GetCSharpResultAt(13, 33, "Math"));
+
+            var fixbody = @"
+        private async Task<int> MathAsync(int num1) 
+        { 
+            await Task.Delay(3000); 
+            return (num1); 
+        }";
+
+            var fixtest = GetFullSource(fixbody);
+            VerifyCSharpFix(test, fixtest);
+        }
+
+        // Adding Async would cause conflict 
+        [Fact(Skip = "https://github.com/dotnet/roslyn-analyzers/issues/1485")]
```

Comment:
You should add RenameAnnotation - https://github.com/dotnet/roslyn-analyzers/issues/1485#issuecomment-378252338